js/src/util/index.js

```diff
@@ -135,20 +135,33 @@ const typeCheckConfig = (componentName, config, configTypes) => {
 }
 
 const isVisible = element => {
-  if (!element) {
+  const eligibleElement = el => el && el.style
+
+  if (!eligibleElement(element)) {
+    return false
+  }
+
+  if (element.tagName !== 'AREA' && element.getClientRects().length === 0) {
     return false
   }
 
-  if (element.style && element.parentNode && element.parentNode.style) {
-    const elementStyle = getComputedStyle(element)
-    const parentNodeStyle = getComputedStyle(element.parentNode)
+  let relevantVisibility
+
+  while (eligibleElement(element)) {
+    const { display, visibility } = getComputedStyle(element)
+
+    if (!relevantVisibility && /visible|hidden/.test(visibility)) {
+      relevantVisibility = visibility
```

Comment:
A bit tricky check IMO .  I saw the extra test and it is valid, however I think it is maybe an edge case, of  bad use

Reply:
According to the CSS spec:

> The visibility property specifies whether the boxes generated by an element are rendered. Invisible boxes still affect layout (set the display property to none to suppress box generation altogether). Values have the following meanings:
>
> - visible
> The generated box is visible.
> - hidden
> The generated box is invisible (fully transparent, nothing is drawn), but still affects layout. Furthermore, descendants of the element will be visible if they have 'visibility: visible'.